R/metric-selection.R

```diff
@@ -58,6 +67,47 @@ contains_survival_metric <- function(mtr_info) {
   any(grepl("_survival", mtr_info$class))
 }
 
+#' @rdname choose_metric
+#' @export
+choose_eval_time <- function(x, metric, eval_time = NULL, call = rlang::caller_env()) {
+  mtr_set <- .get_tune_metrics(x)
+  mtr_info <- tibble::as_tibble(mtr_set)
+
+  if (!contains_survival_metric(mtr_info)) {
+    return(NULL)
+  }
```

Comment:
With this, we don't warn when a user supplies `eval_time` together with a non-survival metric but want to ( #704) - should that be happening in this PR?